lib/facter/test_join.rb

```diff
@@ -0,0 +1,44 @@
+Facter.add(:test_join) do
+  setcode do
+    kernel = Facter.value('kernel')
+
+    case kernel
+    ## linux case
+    #
+    #  Note: the following are additional resources:
+    #
+    #        - https://linux.die.net/man/8/net
+    #        - http://linuxcommand.org/lc3_man_pages/typeh.html
+    #
+    when 'Linux'
+      # check AD is joined
+      Facter::Core::Execution.exec('cd /home; type net &>/dev/null; net ads testjoin &>/dev/null 2>&1; echo $?')
+
+    ## windows case
+    #
+    #  @pattern, an expected output (i.e. windows 7), when there are no active
+    #      network connections.
+    #
+    #  Note: the following are addditional resources:
+    #
+    #        - https://technet.microsoft.com/en-us/library/bb490717.aspx
+    #
+    when 'Windows'
```

Comment:
Can we remove `windows` from the case statement, so that it returns 1 anything other than Linux? My thinking is that would be more appropriate since this module only manages winbind, which is only used on Linux. Also, it looks like according to the [docs](https://technet.microsoft.com/en-us/library/gg651155(v=ws.11).aspx), the `NET USE` command is used for shared resources, such as network drives, rather than the actual domain binding.